src/features/preview.ts

```diff
@@ -433,25 +433,33 @@ export class AsciidocPreview {
     vscode.workspace.openTextDocument(this._resource).then(vscode.window.showTextDocument)
   }
 
-  private async onDidClickPreviewLink (href: string) {
-    let [hrefPath, fragment] = decodeURIComponent(href).split('#')
-    // From Markdown plugin
-    if (hrefPath[0] !== '/') {
+  private resolveDocumentLink (href: string): { hrefPath: string, fragment: string } {
+    let [hrefPath, fragment] = href.split('#').map((c) => decodeURIComponent(c))
+    if (hrefPath.startsWith('file://')) {
+      // Absolute path, this will probably not work on Windows
+      hrefPath = vscode.Uri.parse(hrefPath).fsPath
+    }
+    if (!hrefPath.startsWith('/')) {
+      // Relative path. Resolve relative to the file
       hrefPath = path.join(path.dirname(this.resource.fsPath), hrefPath)
-    } else {
-      // Handle any normalized file paths
-      hrefPath = vscode.Uri.parse(hrefPath.replace('/file', '')).fsPath
     }
+    return { hrefPath: hrefPath, fragment: fragment }
```

Comment:
We could return `{ path: hrefPath, fragment: fragment }`. That way, we don't need to destructure the object when calling `executeCommand` we can directly pass the object:

```ts
vscode.commands.executeCommand('_asciidoc.openDocumentLink', targetResource)
```